spotinst/resource_spotinst_ocean_aks_np_test.go

```diff
@@ -0,0 +1,902 @@
+		)
+	} else {
+		format := testBaselineOceanAKSNPConfig_Create
+		template += fmt.Sprintf(format,
+			clusterMeta.clusterName,
+			clusterMeta.provider,
+			clusterMeta.controllerClusterID,
+			clusterMeta.autoScaler,
+			clusterMeta.health,
+			clusterMeta.scheduling,
+			clusterMeta.taints,
+			clusterMeta.headrooms,
+			clusterMeta.filters,
+		)
+
+	}
+
+	if clusterMeta.variables != "" {
+		template = clusterMeta.variables + "\n" + template
+	}
+
+	log.Printf("Terraform [%v] template:\n%v", clusterMeta.clusterName, template)
+	return template
+}
+
+// region Ocean AKS : Baseline
+func TestAccSpotinstOceanAKSNP_Baseline(t *testing.T) {
+	clusterName := "Terraform-Test-AKS-2-0-Cluster"
+	controllerClusterID := "terraform-aks-2-0-cluster"
+	resourceName := createOceanAKSNPResourceName(clusterName)
+
+	var cluster azure_np.Cluster
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t, "azure") },
+		Providers:    TestAccProviders,
+		CheckDestroy: testOceanAKSNPDestroy,
+
+		Steps: []resource.TestStep{
+			{
+				Config: createOceanAKSNPTerraform(&OceanAKSNPMetadata{
+					clusterName:         clusterName,
+					controllerClusterID: controllerClusterID,
+				}),
+				//ExpectNonEmptyPlan: true,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckOceanAKSNPExists(&cluster, resourceName),
+					testCheckOceanAKSNPAttributes(&cluster, clusterName),
+					resource.TestCheckResourceAttr(resourceName, "name", "Terraform-Test-AKS-2-0-Cluster"),
+					resource.TestCheckResourceAttr(resourceName, "controller_cluster_id", "terraform-aks-2-0-cluster"),
+					resource.TestCheckResourceAttr(resourceName, "aks_cluster_name", "Terraform-Test-AKS-2-0-Do-Not-Delete"),
+					resource.TestCheckResourceAttr(resourceName, "aks_region", "eastus"),
+					resource.TestCheckResourceAttr(resourceName, "aks_resource_group_name", "AutomationResourceGroup"),
+					resource.TestCheckResourceAttr(resourceName, "aks_infrastructure_resource_group_name", "MC_AutomationResourceGroup_Terraform-Test-AKS-2-0-Do-Not-Delete_eastus"),
+					resource.TestCheckResourceAttr(resourceName, "min_count", "1"),
+					resource.TestCheckResourceAttr(resourceName, "max_count", "100"),
+					resource.TestCheckResourceAttr(resourceName, "max_pods_per_node", "30"),
+					resource.TestCheckResourceAttr(resourceName, "enable_node_public_ip", "true"),
+					resource.TestCheckResourceAttr(resourceName, "os_disk_size_gb", "32"),
+					resource.TestCheckResourceAttr(resourceName, "os_disk_type", "Managed"),
+					resource.TestCheckResourceAttr(resourceName, "os_type", "Linux"),
+					resource.TestCheckResourceAttr(resourceName, "os_sku", "Ubuntu"),
+					resource.TestCheckResourceAttr(resourceName, "kubernetes_version", "1.26"),
+					resource.TestCheckResourceAttr(resourceName, "spot_percentage", "50"),
+					resource.TestCheckResourceAttr(resourceName, "fallback_to_ondemand", "false"),
+					//resource.TestCheckResourceAttr(resourceName, "pod_subnet_ids.#", "1"),
```

Comment:
reason for commenting the po-subnet_ids

Reply:
we get error when we use both pod subnet ids and vnet ids